IaC/scripts/cleanup_repo_script.sh

```diff
@@ -0,0 +1,133 @@
+#!/bin/bash
+
+usage () {
+    cat <<EOF
+Usage: $0 [OPTIONS]
+    The following options may be given :
+        --mode              Pass \"latest\" if you want to remove all packages except latest.
+                            Pass \"date\" if you want to remove by specific date
+        --older_than        Pass days older than packages should be removed. For example: 365
+        --repo              Pass name of repository which needed to be cleaned up
+        --component         Pass name of component, supported values: \"laboratory\" \"experimental\" \"testing\" \"all\"
+        --dry-run           Enables dry-run mode, will show what will be removed and what was excluded only
+        --help) usage ;;
+Example $0 --mode=latest
+EOF
+        exit 1
+}
+
+parse_arguments() {
+    if test "$1" = PICK-ARGS-FROM-ARGV
+    then
+        shift
+    fi
+
+    for arg do
+        val=$(echo "$arg" | sed -e 's;^--[^=]*=;;')
+        case "$arg" in
+            --mode=*) MODE="$val" ;;
+            --older_than=*) OLDER_THAN="$val" ;;
+            --repo=*) REPO_LIST="$val" ;;
+            --component=*) REPO_COMPONENT="$val" ;;
+            --dry-run=*) DRY_RUN="$val" ;;
+            --help|*) usage ;;
+        esac
+    done
+}
+
+function main {
+    if [[ "$REPO_COMPONENT" == "all" ]]; then
+        REPO_COMPONENT="testing laboratory experimental"
+    fi
+
+    for REPO in $REPO_LIST; do
+        for COMPONENT in $REPO_COMPONENT; do
+            REPOPATH="/mnt/${REPO}/yum/${COMPONENT}"
+
+            echo "Cleaning repo: ${REPO} & component: ${COMPONENT}"
+
+            if [[ ${MODE} == "date" ]]; then
+                EXCLUDE_COMPONENT=$(echo "$EXCLUDE_LIST" | grep -o "${COMPONENT}=.*" | awk -F '=' '{print $2}')
+                echo "Searching for packages to be excluded across all EL versions"
+
+                for package in $EXCLUDE_COMPONENT; do
+                    find $REPOPATH/ -type f -name "$package.*" > /tmp/excludes_${REPO}_${COMPONENT}
+                done
+                EXCLUDE="$(cat /tmp/excludes_${REPO}_${COMPONENT} | awk -F "/" '{print $NF}')"
+
+                if [[ ${DRY_RUN} == "no" ]]; then
+                    FIND_CMD="rm -f"
```

Comment:
1 this name is not correct as FIND_CMD - means that it it is find command with params
2 I suggest making this global variable